.eslintrc

```diff
@@ -32,31 +32,35 @@
     "no-only-tests",
     "sort-keys-fix",
     "prettier",
-    "promise"
+    "promise",
   ],
   "rules": {
     "react/prop-types": "off",
-    "no-unused-vars": ["error", { "varsIgnorePattern": "^_", "argsIgnorePattern": "^_" }],
+    "no-unsafe-optional-chaining": "off",
```

Comment:
why do we need to turn this off. The PR description mentions that charts won't load without this, but this is only a linting rule and it seems like a good one to keep on.

Reply:
It's like i'm caught between a rock and a hard place. 
This is the line in question. 
https://github.com/AMP-SCZ/dpdash/blob/32550c70c74884d7ea2c1a654c04212b672f31b1/server/services/BarChartTableService/index.js#L98

The optional chaining lint rule and formatter took this out. We need this because it handles an edge case where a study might not have `any` counts for a specific variable.

I'm curious, what do you suggest here, because if we have it as a warning it will always be there, we need it for this edge case, if this rule is on then we get dinged on the github actions, we can be extremely verbose, but do we always want to be that verbose when the language has syntax to reduce the amount of lines of code.

Reply:
I believe es-lint wants us to handle the case where `totalsForStudy` is undefined, so something like this should pass the linter:
```javascript
  _sortedTableRowDataBySite = () => {
    return this.dataBySite
      .map((site) => {
        const { totalsForStudy } = site
        const { count, targetTotal } = totalsForStudy || // We need to do something if the totalsForStudy is undefined, maybe { count: 0, targetTotal: 0 }?
        site.counts[TOTAL_LABEL] = count
        site.targets[TOTAL_LABEL] = targetTotal

        return site
      })
      .sort(this._sortSitesByNameAndTotalsAtBottom)
  }
```

Reply:
Thanks for this @JackVCurtis , for the record this wasn't a disagreement with anyone, it was more to open a discussion of how we would want to handle this and list to everyone's opinions on the matter.

Reply:
[ad6dbfd](https://github.com/AMP-SCZ/dpdash/pull/723/commits/ad6dbfd45e3fb4f3114a08eb996e92b388394b4e)

Reply:
Hmm this didn't work, and the error keeps expanding. It's not in scope of this ticket so i'll make a bug ticket for this, as it requires more attention.

Reply:
https://github.com/AMP-SCZ/dpdash/issues/725